### DISCUSSION TITLE
JSON formatting after you write into specific file

### DISCUSSION THREAD

**Discussion by hvaandres** [cypress-io/cypress#14905] [Questions and Help] [answered]
I'm currently writing into a JSON file every single time that I'm using new login credentials. To make sure we have those emails to make sure we can do an automated process of verifying this information. 

The way that I'm doing it is by using cypress.writing():

```
cy.get('#email')
      .type(email).writeFile('cypress/fixtures/users.json', { name: fullName, email: email}, {flag: 'a+'})

```

However, every single time that I run my script, I will have this JSON file as a result:

```
{
  "name": "MockaData Testing",
  "email": "29nLcQ@aharo.com"
}{
  "name": "MockaData Testing",
  "email": "XPJI1b@aharo.com"
}{
  "name": "MockaData Testing",
  "email": "wjkqon@aharo.com"
}

```

As you can see the formatting is not correct. What can I do to fix this issue?

**Comment by jennifer-shehane** [comment, marked as answer]
@hvaandres What is the final JSON file you are expecting to be written?

**Reply by bahmutov** [reply]
Yeah @hvaandres how would you append JSON objects to each other? I think you want to keep the list right? So I think you would do something like this:
```js
const usersFilename = './users.json'
beforeEach(() => {
  // start with an empty list
  cy.writeFile(usersFilename, '[]')
})
it('works', () => {
  ...
  cy.get('#email').type(email)
  // now let's read the existing list of users
  // insert new object and write the updated list
  cy.readFile(usersFilename).then(users => {
    users.push({
       name: fullName,
       email
    })
    cy.writeFile(usersFilename, users)
  })
})
```

**Reply by hvaandres** [reply]
Thank you @bahmutov you solved my issue
